devimint/src/main.rs

```diff
@@ -1397,11 +1396,61 @@ pub async fn recoverytool_test(dev_fed: DevFed) -> Result<()> {
     } = dev_fed;
     let data_dir = env::var("FM_DATA_DIR")?;
 
-    let peg_in_values = HashSet::from([12_345_000, 23_456_000, 34_567_000]);
-    for peg_in_value in &peg_in_values {
-        fed.pegin(*peg_in_value).await?;
+    let mut fed_utxos_sats = HashSet::from([12_345_000, 23_456_000, 34_567_000]);
+    for sats in &fed_utxos_sats {
+        fed.pegin(*sats).await?;
     }
-    let total_pegged_in = peg_in_values.iter().sum::<u64>();
+
+    // Initiate a withdrawal to verify the recoverytool recognizes change outputs
+    let withdrawal_address = bitcoind.get_new_address().await?;
+    let withdraw_res = cmd!(
+        fed,
+        "withdraw",
+        "--address",
+        &withdrawal_address,
+        "--amount",
+        "5000 sat"
+    )
+    .out_json()
+    .await?;
+
+    let fees_sat = withdraw_res["fees_sat"]
+        .as_u64()
+        .expect("withdrawal should contain fees");
+    let txid: Txid = withdraw_res["txid"]
+        .as_str()
+        .expect("withdrawal should contain txid string")
+        .parse()
+        .expect("txid should be parsable");
+    let tx_hex = poll("Waiting for transaction in mempool", None, || async {
+        bitcoind
+            .get_raw_transaction(&txid)
+            .await
+            .context("getrawtransaction")
+            .map_err(ControlFlow::Continue)
+    })
+    .await
+    .expect("withdrawal tx failed to reach mempool");
+
+    let tx = bitcoin::Transaction::consensus_decode_hex(&tx_hex, &Default::default())?;
+    assert_eq!(tx.input.len(), 1);
+    assert_eq!(tx.output.len(), 2);
+
+    let withdrawal_address = bitcoin::Address::from_str(&withdrawal_address)?;
+    let change_output = tx
+        .output
+        .iter()
+        .find(|o| o.to_owned().script_pubkey != withdrawal_address.script_pubkey())
+        .expect("withdrawal must have change output");
+    assert!(fed_utxos_sats.insert(change_output.value));
+
+    // Remove the utxo consumed from the withdrawal tx
+    let total_output_sats = tx.output.iter().map(|o| o.value).sum::<u64>();
+    let input_sats = total_output_sats + fees_sat;
+    assert!(fed_utxos_sats.remove(&input_sats));
+
+    let total_fed_sats = fed_utxos_sats.iter().sum::<u64>();
+    fed.generate_epochs(2).await?;
```

Comment:
Great eye!

I think there's an off-by-one error since we don't recognize the change output generating only one epoch.

I created an issue and will start looking into it. I'll also rename in the fix.
https://github.com/fedimint/fedimint/issues/3884